### PR TITLE
add the recipe cairomm

### PR DESCRIPTION
**Pull request by vgauthier**

### CODE REVIEW

**Comment by jakirkham** [comment]
Please make the same sort of changes here as we did to PR ( https://github.com/conda-forge/staged-recipes/pull/1083 ).

**Comment by jakirkham** [comment]
Going to try CircleCI again as it didn't start.

File: recipes/cairomm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - pkg-config
+    - toolchain
+  run:
+    - cairo
+    - libsigcpp
+    - libpng
+
+test:
+    commands:
+        - test -f $PREFIX/lib/libcairomm-1.0.a           # [linux]
```
Comment (jakirkham) [inline recipes/cairomm/meta.yaml]: Static library should be on both platforms right?

Also, please test for the Linux shared library.

**Comment by jakirkham** [comment]
Please make sure to add `[skip appveyor]` to your commit messages as this doesn't build on Windows.

**Comment by vgauthier** [comment]
as opposed to #1083 the code is necessary to correctly setup the compilation with clang on osx

```bash
if [[ $(uname) == Darwin ]]; then
    # for Mac OSX
    export CC=clang
    export CXX=clang++
    export MACOSX_VERSION_MIN="10.7"
    export MACOSX_DEPLOYMENT_TARGET="${MACOSX_VERSION_MIN}"
    export CXXFLAGS="${CXXFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
    export CXXFLAGS="${CXXFLAGS} -stdlib=libc++"
    export LDFLAGS="${LDFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
    export LDFLAGS="${LDFLAGS} -stdlib=libc++ -lc++"
    export LINKFLAGS="${LDFLAGS}"
fi

```

**Comment by jakirkham** [comment]
That is all done for you by the `toolchain`. See the [`activate.sh`]( https://github.com/conda-forge/toolchain-feedstock/blob/9ed1c8d6b58e4186684ed1e72a167fcabf3a1383/recipe/activate.sh ).

**Comment by vgauthier** [comment]
ok for activate.sh I'll make the change, I didn't know that

File: recipes/cairomm/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/cairomm/meta.yaml]: Half done. 😉

Comment (jakirkham) [inline recipes/cairomm/meta.yaml]: Done.

**Comment by jakirkham** [comment]
Still confused why CircleCI is not running on your PRs. Have you blocked the OAuth App?

File: recipes/cairomm/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# FIXME: This is a hack to make sure the environment is activated.
+# The reason this is required is due to the conda-build issue
+# mentioned below.
+#
+# https://github.com/conda/conda-build/issues/910
+#
+source activate "${CONDA_DEFAULT_ENV}"
+
+# Setup the building
+./configure --prefix=${PREFIX}  || { cat config.log; exit 1; }
```
Comment (jakirkham) [inline recipes/cairomm/build.sh]: Probably need a flag like `--enable-static`. Would add `--enable-shared` for good measure.

**Comment by vgauthier** [comment]
nop, I haven't block the OAuth App

File: recipes/cairomm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    url: https://www.cairographics.org/releases/cairomm-{{ version }}.tar.gz
+    {{ checktype }}: {{ checksum }}
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  # Packages required to build the package. python and numpy must be
+  # listed explicitly if they are required.
```
Comment (jakirkham) [inline recipes/cairomm/meta.yaml]: nit: please drop this comment.

**Comment by jakirkham** [comment]
One nit. Otherwise LGTM.

**Comment by vgauthier** [comment]
@jakirkham do you want me to revoke my CircleCI token and restart with a clean slate ?

**Comment by jakirkham** [comment]
Kinda. I'm mainly just worried that something funky is going on with your CircleCI settings somehow as it seems restricted to your PRs. Would it be alright for you to try?

**Comment by vgauthier** [comment]
@jakirkham it seems revoke my token and recreate it, work!!

**Comment by jakirkham** [comment]
Awesome! Glad we got that straightened out. Thanks for experimenting with this with me, @vgauthier.

**Comment by jakirkham** [comment]
Thanks again for your hardwork here, @vgauthier. Let's get this out there. 😄

**Pull request merged**